ledger/accountdb.go

```diff
@@ -3288,6 +3478,31 @@ func (w accountsSQLWriter) updateResource(addrid int64, aidx basics.CreatableInd
 	return
 }
 
+func (w accountsSQLWriter) upsertKvPair(key string, value string) error {
+	// NOTE! If we are passing in `string`, then for `BoxKey` case,
+	// we might contain 0-byte in boxKey, coming from uint64 appID.
+	// The consequence would be DB key write in be cut off after such 0-byte.
```

Comment:
@ahangsu Did you investigate why, or just discover the issue, cast to `[]byte` and move on?

Reply:
yes, this is for if we are passing in string, then imaging the box prefix containing byte 0, the write to DB will stop and "null-terminated", while converting them to string get around the problem.

Reply:
I'm currently reasonably settled on converting values to `[]byte`, but leaving `key` as `string`, so this comment remains. I did not convert the keys to []byte because we'd need to convert them to string at various places where we use them as keys in maps.

https://github.com/algorand/go-algorand/pull/4686

Reply:
Resolving - Since there's been no further discussion here, let's resolve with the as-is inline comments documenting the concern.